### ISSUE TITLE
Implement RStudio Connection Contract

### ISSUE THREAD

**Issue by krlmlr** [r-dbi/dbi3#11]
Link: https://rstudio.github.io/rstudio-extensions/connections-contract.html

Should be possible by changing the implementation of the generic to

```r
setGeneric("dbConnect",
  def = function(drv, ...) {
    con <- standardGeneric("dbConnect")
    register_con_with_rstudio(con)
    con
  },
  valueClass = "DBIConnection"
)
```

instead of the current implementation:

```r
setGeneric("dbConnect",
  def = function(drv, ...) standardGeneric("dbConnect"),
  valueClass = "DBIConnection"
)
```

Similar for `dbDisconnect()`. I'm not sure about the `connectionUpdated()` entry point, because DBI currently doesn't provide notification.

Implementation sketch by @edgararuiz: https://gist.github.com/edgararuiz/0533b3daf70a41b70acd75c1e6a2b7a3.

Happy to review a PR. CC @nwstephens @jimhester.

**Comment by edgararuiz** [comment]
Right, `connections` is an external package the wraps `DBI` so it can open, update and close the Connections pane.  The idea would be integrate the routines from `connections` into `DBI`.  `connections` has a dependency on `rscontract` (https://github.com/rstudio/rscontract).  The `rscontract` package is an interface tool that makes it easier to create and customize the Contract.  I can see that eventually, `DBI` would have that same dependency.
